integration_tests/features/StressTest.feature

```diff
@@ -1,39 +1,72 @@
+@long-running
 Feature: Stress Test
 
-@long-running
-Scenario: Stress Test
-    Given I have a seed node NODE1
-    And I have a seed node NODE2
-    And I have 1 base nodes connected to all seed nodes
-    And I have wallet WALLET_A connected to seed node NODE1
-    And I have wallet WALLET_B connected to seed node NODE2
-    And I have a merge mining proxy PROXY connected to NODE1 and WALLET_A
-    # We need to ensure the coinbase lock heights are reached; mine enough blocks
-    # The following line is how you could mine directly on the node
-    When I merge mine 8 blocks via PROXY
+    @long-running
+    Scenario Outline: Ramped Stress Test
+        Given I have a seed node NODE1
+        And I have stress-test wallet WALLET_A connected to the seed node NODE1 with broadcast monitoring timeout 30
+        And I have a merge mining proxy PROXY connected to NODE1 and WALLET_A
+        # We mine some blocks before starting the other nodes to avoid a spinning sync state when all the nodes are at height 0
+        When I merge mine 6 blocks via PROXY
+        And I have a seed node NODE2
+        And I have <NumNodes> base nodes connected to all seed nodes
+        And I have stress-test wallet WALLET_B connected to the seed node NODE2 with broadcast monitoring timeout 30
+        # There need to be at least as many mature coinbase UTXOs in the wallet coin splits required for the number of transactions
+        When I merge mine <NumCoinsplitsNeeded> blocks via PROXY
+        Then all nodes are at current tip height
+        When I wait for wallet WALLET_A to have at least 5100000000 tari
+
+        Then I coin split tari in wallet WALLET_A to produce <NumTransactions> UTXOs of 5000 uT each with fee_per_gram 20 uT
+        When I merge mine 3 blocks via PROXY
+        When I merge mine <NumCoinsplitsNeeded> blocks via PROXY
+        Then all nodes are at current tip height
+        Then wallet WALLET_A detects all transactions as Mined_Confirmed
+        When I send <NumTransactions> transactions of 1111 uT each from wallet WALLET_A to wallet WALLET_B at fee_per_gram 20
+        # Mine enough blocks for the first block of transactions to be confirmed.
+        When I merge mine 4 blocks via PROXY
```

Comment:
In my testing for the 10,000 transaction scenario, the `Then while merge mining via PROXY...` step timed out every time when it wants to detect the 1st transaction in the list as mined. Investigating the logs, only 13 consecutive full blocks were mined, where theoretically 16 are needed. The specific transaction in question was still in the mempool. Thus, mine more blocks before starting the detection.
```suggestion
        When I merge mine <NumCoinsplitsNeeded> blocks via PROXY
```

Reply:
The statement after this mines a block every time it times out so it should catch up. I feel that is more representative of reality compared to just mining a huge number of blocks in one go.

Reply:
Sadly, in my tests, it did not catch up, but timed out, hence my comment. 

To pause between mining blocks is not a bad idea at all, which could be done in a separate function, it is just that in this case not enough blocks have been mined before the detection function times out.

Reply:
Then something is broken because this statement:
`Then while merge mining via PROXY all transactions in wallet WALLET_A are found to be Mined_Confirmed` should mine a block every 5 seconds.

Reply:
But let us wait and see if it passes in `ci/circleci: run-integration-tests`; if so all good then it must be something in my desktop environment.